client/src/linkManager.ts

```diff
@@ -414,17 +464,22 @@ export class linkManager extends LuigiClientBase {
         relative: path[0] !== '/'
       })
     };
+
     helpers.sendPostMessageToLuigiCore(pathExistsMsg);
-    return pathExistsPromises[currentId];
+
+    return pathExistsPromises[`${currentId}`];
   }
 
   /**
    * Checks if there is one or more preserved views. You can use it to show a **back** button.
    * @memberof linkManager
    * @returns {boolean} indicating if there is a preserved view you can return to
    */
-  hasBack() {
-    return !!this.currentContext.internal.modal || this.currentContext.internal.viewStackSize !== 0;
+  hasBack(): boolean {
+    return (
+      !!(lifecycleManager.currentContext.internal as Record<string, any>)['modal'] ||
+      (lifecycleManager.currentContext.internal as Record<string, any>)['viewStackSize'] !== 0
```

Comment:
**this.currentContext.internal.modal**
vs
**lifecycleManager.currentContext.internal**

Importing **lifecycleManager** in order to reuse its defined **currentContext** might work, but seems like a hard to maintain pattern in the future as we always need to think about breaking linkManager when making changes on lifecycleManger. Rather have it in this own class.